homeassistant/components/rest/binary_sensor.py

```diff
@@ -86,15 +102,34 @@ def setup_platform(hass, config, add_entities, discovery_info=None):
     if rest.data is None:
         raise PlatformNotReady
 
-    # No need to update the sensor now because it will determine its state
-    # based in the rest resource that has just been retrieved.
-    add_entities([RestBinarySensor(hass, rest, name, device_class, value_template)])
+    add_entities(
+        [
+            RestBinarySensor(
+                hass,
+                rest,
+                name,
+                device_class,
+                value_template,
+                force_update,
+                resource_template,
+            )
+        ]
+    )
 
 
-class RestBinarySensor(BinarySensorEntity):
+class RestBinarySensor(BinarySensorDevice):
```

Comment:
We should revert this change. `BinarySensorEntity` is the correct class that we should inherit.

Reply:
Hello Martin,

Looks like I misunderstood #34462 . 

As I see now in docs http://dev-docs.home-assistant.io/en/master/api/components.html
BinarySensorDevice is now deprecated and one need to use BinarySensorEntity. Is it correct?

Do I need to  make one more commit to rename BinarySensorDevice to BinarySensorEntity?

Reply:
Please open a new PR and change to `BinarySensorEntity`. Thanks!

Reply:
Created PR to adjust this: <https://github.com/home-assistant/core/pull/37054>
To ensure it is in time for the beta cut.

Reply:
@frenck Thank You. I understood that I need to open a new PR but didn't realize how to do it. You saved me. ;)